test/unit/preprocessing/pass_foreign_theory_rewrite_white.h

```diff
@@ -0,0 +1,111 @@
+/*********************                                                        */
+/*! \file pass_foreign_theory_rewrite.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aina Niemetz, Mathias Preiner, Andres Noetzli
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Unit tests for Foreign Theory Rerwrite prepricessing pass
+ ** Unit tests for Foreign Theory Rerwrite prepricessing pass
+ **/
+
+#include <cxxtest/TestSuite.h>
+
+#include <iostream>
+#include <vector>
+
+#include "context/context.h"
+#include "expr/node.h"
+#include "expr/node_manager.h"
+#include "preprocessing/passes/foreign_theory_rewrite.h"
+#include "preprocessing/preprocessing_pass_registry.h"
+#include "smt/preprocessor.h"
+#include "smt/process_assertions.h"
+#include "smt/smt_engine.h"
+#include "smt/smt_engine_scope.h"
+#include "smt/smt_statistics_registry.h"
+#include "test_utils.h"
+#include "theory/booleans/circuit_propagator.h"
+#include "theory/rewriter.h"
+
+using namespace CVC4;
+using namespace CVC4::preprocessing;
+using namespace CVC4::preprocessing::passes;
+using namespace CVC4::theory;
+using namespace CVC4::theory::booleans;
+using namespace CVC4::smt;
+
+class ForeignTheoryRewriteWhite : public CxxTest::TestSuite
+{
+ public:
+  ForeignTheoryRewriteWhite() {}
+
+  void setUp() override
+  {
```

Comment:
This seems really excessive to test this theory preprocessing pass.  Either we should make this a more generic utility, or you can just test a static function of your preprocessing pass.

I would prefer the latter.